molviewspec/molviewspec/mvsx_converter.py

```
@@ -0,0 +1,342 @@
+                        full_url = urljoin(base_url, uri)
+                    else:
+                        full_url = uri
+
+                    logger.info(f"Downloading external resource: {full_url}")
+
+                    # Extract filename from URL path
+                    filename = os.path.basename(parsed_uri.path)
+                    if not filename:
+                        # Generate a name if none is available
+                        filename = f"resource_{len(uri_mapping)}"
+
+                    # Ensure we have a unique filename
+                    local_path = os.path.join(temp_dir, filename)
+                    counter = 1
+                    while os.path.exists(local_path):
+                        name, ext = os.path.splitext(filename)
+                        filename = f"{name}_{counter}{ext}"
+                        local_path = os.path.join(temp_dir, filename)
+                        counter += 1
+
+                    # Download the file
+                    try:
+                        urllib.request.urlretrieve(full_url, local_path)
+                        # Add to the mapping
+                        uri_mapping[uri] = filename
+                    except Exception as e:
+                        error_msg = f"Failed to download {full_url}: {e}"
+                        logger.error(error_msg)
+                        download_errors.append(error_msg)
+
+                elif not is_external:
+                    # Local file reference
+                    local_file_path = input_path.parent / uri
+                    if local_file_path.exists():
+                        logger.info(f"Found local file: {local_file_path}")
+                        # Keep the same relative path in the archive
+                        uri_mapping[uri] = uri
+                    else:
+                        logger.warning(
+                            f"Local file not found: {local_file_path}"
+                        )
```

Should this raise an error as well? Maybe I'm missing something.